pyteal/ast/subroutine_test.py

```diff
@@ -1138,177 +1141,202 @@ def mySubroutine(a):
+
+    assert isinstance(declaration.body, pt.Seq)
+    assert len(declaration.body.args) == 11
+
+    for i in range(10):
+        assert isinstance(declaration.body.args[i], pt.ScratchStackStore)
+
+    for i in range(10):
+        assert declaration.body.args[i].slot is argSlots[-i - 1]
+
+    options.setSubroutine(definition)
+    expected, _ = pt.Seq(declaration.body.args[:10] + [return_value]).__teal__(options)
+
+    actual, _ = declaration.__teal__(options)
+    options.setSubroutine(None)
+    assert actual == expected
+
+
+@pytest.mark.parametrize("return_type, return_value", SUBROUTINE_VAR_ARGS_CASES)
+def test_evaluate_subroutine_frame_pt_version(
+    return_type: pt.TealType, return_value: pt.Expr
+):
+    def mySubroutine_no_arg():
+        return return_value
+
+    def mySubroutine_arg_1(a1):
+        return return_value
+
+    def mySubroutine_arg_2(a1, a2):
+        return return_value
+
+    def mySubroutine_arg_10(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10):
+        return return_value
+
+    for subr in [
+        mySubroutine_no_arg,
+        mySubroutine_arg_1,
+        mySubroutine_arg_2,
+        mySubroutine_arg_10,
+    ]:
+        subr = cast(Callable[..., pt.Expr], subr)
+        definition = pt.SubroutineDefinition(subr, return_type)
+        evaluate_subroutine = SubroutineEval.fp_evaluator()
 
-        evaluate_subroutine = SubroutineEval.normal_evaluator()
         declaration = evaluate_subroutine(definition)
 
         assert isinstance(declaration, pt.SubroutineDeclaration)
         assert declaration.subroutine is definition
 
-        assert declaration.type_of() == returnValue.type_of()
-        assert declaration.has_return() == returnValue.has_return()
+        assert declaration.type_of() == return_value.type_of()
+        assert declaration.has_return() == return_value.has_return()
 
         assert isinstance(declaration.body, pt.Seq)
-        assert len(declaration.body.args) == 11
-
-        for i in range(10):
-            assert isinstance(declaration.body.args[i], pt.ScratchStackStore)
+        assert len(declaration.body.args) == 2
 
-        for i in range(10):
-            assert declaration.body.args[i].slot is argSlots[-i - 1]
+        assert isinstance(declaration.body.args[0], Proto)
+        assert isinstance(declaration.body.args[1], type(return_value))
 
-        options.setSubroutine(definition)
-        expected, _ = pt.Seq(declaration.body.args[:10] + [returnValue]).__teal__(
-            options
+        options_v8.setSubroutine(definition)
+        expected, _ = pt.Seq([declaration.body.args[0], return_value]).__teal__(
+            options_v8
         )
 
-        actual, _ = declaration.__teal__(options)
-        options.setSubroutine(None)
-        assert actual == expected
+        actual, _ = declaration.__teal__(options_v8)
+        options_v8.setSubroutine(None)
+        with pt.TealComponent.Context.ExprEqualityContext():
```

Comment:
I don't think this is necessary/desired here

Reply:
just removed the context